### ISSUE TITLE
[Bug]: DC0007/DC0008 never fire for interface and controladdin objects

### ISSUE THREAD

**Issue by Arthurvdv** [ALCops/Analyzers#451]
### Analyzer

DocumentationCop

### Issue type

False negative (diagnostic doesn't fire when it should)

### Rule ID

DC0007 / DC0008

### Description

While fixing #438 (DC0004 false positive on interface procedures), a related latent bug surfaced in `ObjectRequiresDocumentation`:

The analyzer registers for the `Interface` and `ControlAddIn` symbol kinds, but the analysis callback bails out early:

```csharp
if (ctx.Symbol is not IApplicationObjectTypeSymbol appObjectTypeSymbol)
{
    return;
}
```

In the NAV SDK, `IInterfaceTypeSymbol` and `IControlAddInTypeSymbol` implement only `IObjectTypeSymbol`, **not** `IApplicationObjectTypeSymbol` (the two interfaces are siblings). So interface and controladdin objects **never** raise DC0007/DC0008, even though the registration suggests they were intended to be covered.

### Question for the original author

What is the intended object-level behavior?

1. Should public interfaces raise DC0007 (and internal ones DC0008)? Interface signatures are the cross-extension API contract, so this seems consistent with the rule intent.
2. Should controladdin objects be covered too, or removed from the registration? For DC0004 we decided to skip controladdin *procedures* entirely (JS-implemented contract), but the object itself is referenced cross-extension via `usercontrol`.

### Additional context

- Root cause analysis and the layered-resolution fix pattern are documented in `.github/instructions/dc0004-procedure-requires-documentation.instructions.md` (added by the PR fixing #438).
- Related: #438

**Comment by MODUSCarstenScholling** [comment]
Yes. Public interfaces should throw **DC0007** and internal interfaces should throw **DC0008**. An interface specifies an AL API contract and its signatures are used by implementations and dependent extensions. Therefore, object level documentation should describe the purpose and intended use of the contract in accordance with DC0004/DC0006 procedure behavior.

Addins keep control in the registration report **DC0007**. The controladdin object is part of the AL-facing API and is referenced by consumers via `usercontrol`. So it should require object-level XML documentation. And we should reconsider not including ControlAddIn procedures completely (see #438 ), as the ControlAddIn is part of the cross extension API via usercontrol.

**Comment by MODUSCarstenScholling** [comment]
I'll catch this in the next week.

**Comment by MODUSCarstenScholling** [comment]
@Arthurvdv I would like to reintroduce the DC0004 for controladdin procedures/events. They are part of the (public) API. What do you think?

**Comment by Arthurvdv** [comment]
@MODUSCarstenScholling I don't see any reason why not 🤷

The only thing could be a `event AddInReady()` and/or `event ControlReady()` which is kinda a boilerplate lifecycle event that every controladdin must declare _(fires when the JS control has loaded and is ready)_.

There's no reliable list of "boilerplate lifecycle events" that I know of that we can easily skip? If the ControlReady is self-evident, writing a one-line ```/// <summary>Raised when the control is ready.</summary>``` is minimal effort and keeps the rule simple and predictable would be my take.
